src/core/introForElement.js

```diff
@@ -146,35 +130,31 @@ export default function introForElement(targetElm, group) {
           disableInteraction = this._options.disableInteraction;
         }
 
-        introItems[nextStep] = {
-          element: currentElement,
-          title: currentElement.getAttribute("data-title") || "",
-          intro: currentElement.getAttribute("data-intro"),
-          step: nextStep + 1,
-          tooltipClass: currentElement.getAttribute("data-tooltipclass"),
-          highlightClass: currentElement.getAttribute("data-highlightclass"),
-          position:
-            currentElement.getAttribute("data-position") ||
-            this._options.tooltipPosition,
-          scrollTo:
-            currentElement.getAttribute("data-scrollto") ||
-            this._options.scrollTo,
-          disableInteraction,
-        };
+        if (step > 0) {
```

Comment:
This is a scary function to change mainly because we don't have any tests for it.

Reply:
Yeah, the point is to use a single loop, instead of two. As such, existing tests already cover this.

Reply:
Not sure if we have tests for this? there are two ways of adding the steps, using the data-* attributes and a JSON object. I don't think we have any tests for the data-* attr approach.

Reply:
I just started refactoring this function and adding some tests. I will ping you once it's merged.

Reply:
WIP https://github.com/usablica/intro.js/pull/1295